analyzers/src/SonarAnalyzer.CFG/ShimLayer/TypeInfoExtensions.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Tunnel Vision Laboratories, LLC. All Rights Reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace StyleCop.Analyzers.Lightup
+{
+    using System;
+    using Microsoft.CodeAnalysis;
+    using static System.Linq.Expressions.Expression;
+
+    public static class TypeInfoExtensions
+    {
+        private static readonly Func<TypeInfo, NullabilityInfo> ConvertedNullabilityAccessor = CreateNullabilityAccessor(nameof(ConvertedNullability));
+        private static readonly Func<TypeInfo, NullabilityInfo> NullabilityAccessor = CreateNullabilityAccessor(nameof(Nullability));
+
+        public static NullabilityInfo ConvertedNullability(this TypeInfo typeInfo) =>
+            ConvertedNullabilityAccessor(typeInfo);
+
+        public static NullabilityInfo Nullability(this TypeInfo typeInfo) =>
+            NullabilityAccessor(typeInfo);
+
+        private static Func<TypeInfo, NullabilityInfo> CreateNullabilityAccessor(string propertyName)
+        {
+            var typeInfoType = typeof(TypeInfo);
+            var property = typeInfoType.GetProperty(propertyName);
```

Comment:
Can be merged, we don't need to reuse `typeInfoType`
```suggestion
            var typeInfoType = typeof(TypeInfo).GetProperty(propertyName);
```